### PR TITLE
eNikshay repeaters report

### PR DESCRIPTION
**Pull request by proteusvacuum**
@benrudolph @dannyroberts 

We need a way of auditing our repeaters for eNikshay at a more custom level than is currently available in the repeat records report. 

It would be great to get this in by EOD Friday since we actually need to send this report along to 99DOTS. 

@mkangia FYI

### CODE REVIEW

File: corehq/apps/repeaters/dbaccessors.py [hunk 1]
```diff
@@ -83,14 +75,46 @@ def get_paged_repeat_records(domain, skip, limit, repeater_id=None, state=None):
...
+    from .models import RepeatRecord
+    kwargs = {
+        'include_docs': True,
+        'reduce': False,
+        'descending': True,
+    }
+    kwargs.update(_get_startkey_endkey_all_records(domain, repeater_id, state))
+
+
+    for doc in paginate_view(
```
Comment (dimagimon) [inline corehq/apps/repeaters/dbaccessors.py]: E303 too many blank lines (2)

File: custom/enikshay/reports/repeaters.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            escape(record.failure_reason) if not record.succeeded else None,
+            payload,
+        ]
+        return row
+
+    def _get_person_id_link(self, record):
+        try:
+            person_id = get_person_case(self.domain, record.payload_id)
+            return '<a href="{url}" target="_blank">{case_id}</a>'.format(
+                    url=reverse('case_details', args=[self.domain, person_id]),
```
Comment (dimagimon) [inline custom/enikshay/reports/repeaters.py]: E126 continuation line over-indented for hanging indent

File: custom/enikshay/reports/repeaters.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                    case_id=person_id
+                )
```
Comment (dimagimon) [inline custom/enikshay/reports/repeaters.py]: E121 continuation line under-indented for hanging indent

File: corehq/apps/repeaters/dbaccessors.py [hunk 4]
```diff
@@ -83,14 +75,46 @@ def get_paged_repeat_records(domain, skip, limit, repeater_id=None, state=None):
         kwargs['endkey'] = [domain, None, state]
         kwargs['startkey'] = [domain, None, state, {}]
     elif not repeater_id and not state:
-        kwargs['endkey'] = [domain]
-        kwargs['startkey'] = [domain, {}]
+        kwargs['endkey'] = [domain, None]
```
Comment (proteusvacuum) [inline corehq/apps/repeaters/dbaccessors.py]: @benrudolph for some reason this was returning all the records twice if the `None` wasn't there. I added a test for this case.

Comment (benrudolph) [inline corehq/apps/repeaters/dbaccessors.py]: thanks sorry about that. must've missed that case when i updated the view

File: custom/enikshay/reports/repeaters.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+            DataTablesColumn(_('Payload')),
+        ]
+
+        return DataTablesHeader(*columns)
+
+    def _make_row(self, record):
+        try:
+            payload = record.get_payload(save_failure=False)
+        except ENikshayCaseNotFound as error:
+            payload = "Error: {}".format(error)
```
Comment (benrudolph) [inline custom/enikshay/reports/repeaters.py]: nit: not sure if this ever needs to support unicode

**Review by benrudolph — APPROVED** [review]
looks good!

**Pull request merged**